File: test-connection.ts
import pg from 'pg';
import mysql from 'mysql2/promise';
import fs from 'fs';

const envContent = fs.readFileSync('.env', 'utf-8');
const env: Record<string, string> = {};
envContent.split('\n').forEach(line => {
  const [key, value] = line.split('=');
  if (key && value) {
    env[key.trim()] = value.trim();
  }
});

const DB_TYPE = env.DB_TYPE || 'mysql';
const DB_CONFIG = {
  host: env.DB_HOST || 'localhost',
  port: parseInt(env.DB_PORT || (DB_TYPE === 'postgres' ? '5432' : '3306')),
  user: env.DB_USER || 'root',
  password: env.DB_PASSWORD || 'yj1234',
  database: env.DB_NAME || 'text2sql_demo',
  ssl: env.DB_SSL === 'true' ? { rejectUnauthorized: false } : undefined,
};

async function testPostgres() {
  console.log('Testing PostgreSQL connection...');
  console.log('Config:', { ...DB_CONFIG, password: '***' });
  const pool = new pg.Pool(DB_CONFIG);
  const client = await pool.connect();
  try {
    const result = await client.query('SELECT COUNT(*) FROM products');
    console.log('\n✅ Connection successful!');
    console.log(`Number of products in database: ${result.rows[0].count}`);
    const tables = await client.query(`SELECT table_name FROM information_schema.tables WHERE table_schema = 'public'`);
    console.log('\nTables found:');
    tables.rows.forEach(row => console.log(`  - ${row.table_name}`));
  } catch (error) {
    console.error('\n❌ Connection failed:', (error as Error).message);
    throw error;
  } finally {
    client.release();
    await pool.end();
  }
}

async function testMySQL() {
  console.log('Testing MySQL connection...');
  console.log('Config:', { ...DB_CONFIG, password: '***' });
  const connection = await mysql.createConnection(DB_CONFIG);
  try {
    const [rows] = await connection.query('SELECT COUNT(*) as count FROM products');
    console.log('\n✅ Connection successful!');
    console.log(`Number of products in database: ${(rows as any)[0].count}`);
    const [tables] = await connection.query(`SELECT TABLE_NAME FROM information_schema.TABLES WHERE TABLE_SCHEMA = ?`, [DB_CONFIG.database]);
    console.log('\nTables found:');
    (tables as any).forEach((row: any) => console.log(`  - ${row.TABLE_NAME}`));
  } catch (error) {
    console.error('\n❌ Connection failed:', (error as Error).message);
    throw error;
  } finally {
    await connection.end();
  }
}

if (DB_TYPE === 'postgres' || DB_TYPE === 'supabase') {
  testPostgres();
} else {
  testMySQL();
}

File: test_connection.py
#!/usr/bin/env python3
import os
import psycopg2
from psycopg2.extras import RealDictCursor
from dotenv import load_dotenv

load_dotenv()

DB_HOST = os.getenv("DB_HOST", "localhost")
DB_PORT = int(os.getenv("DB_PORT", "5432"))
DB_USER = os.getenv("DB_USER", "postgres")
DB_PASSWORD = os.getenv("DB_PASSWORD", "")
DB_NAME = os.getenv("DB_NAME", "postgres")
DB_SSL = os.getenv("DB_SSL", "false").lower() == "true"

def main():
    print("Testing database connection...")
    print(f"Host: {DB_HOST}")
    print(f"Port: {DB_PORT}")
    print(f"User: {DB_USER}")
    print(f"Database: {DB_NAME}")
    print(f"SSL: {DB_SSL}")
    print()

    try:
        sslmode = "require" if DB_SSL else "disable"
        conn = psycopg2.connect(
            host=DB_HOST,
            port=DB_PORT,
            user=DB_USER,
            password=DB_PASSWORD,
            dbname=DB_NAME,
            sslmode=sslmode
        )

        print("✅ Connected successfully!")
        print()

        cursor = conn.cursor(cursor_factory=RealDictCursor)
        cursor.execute("SELECT table_name FROM information_schema.tables WHERE table_schema = 'public'")
        tables = cursor.fetchall()

        if tables:
            print(f"Found {len(tables)} tables:")
            for table in tables:
                print(f"  - {table['table_name']}")
        else:
            print("No tables found in public schema.")

        print()
        print("✅ Database connection test passed!")

        cursor.close()
        conn.close()

    except Exception as e:
        print(f"❌ Connection failed: {e}")
        exit(1)

if __name__ == "__main__":
    main()
